### PR TITLE
Update nvcomp to 4.0.1

### PR DESCRIPTION
**Pull request by vuule**
Update the nvCOMP version used for compression/decompression to 4.0.1.

See also:

https://github.com/rapidsai/rapids-cmake/pull/633
https://github.com/rapidsai/cudf/pull/16076

Depends on https://github.com/rapidsai/kvikio/pull/450.

### CODE REVIEW

File: cpp/cmake/rapids_config.cmake
Comment (madsbk) [inline cpp/cmake/rapids_config.cmake:16]:
```suggestion
```

Remeber to remove before PR merge

**Review by madsbk — APPROVED** [review]
Looks good, thanks @vuule.
PS: I have marked the PR `DO NOT MERGE` as a reminder that we need revert the  `rapids-cmake-repo` change before merging :)

File: cpp/cmake/rapids_config.cmake
Comment (msarahan) [reply cpp/cmake/rapids_config.cmake:16]: Note (more for me than anything): this can be reverted when https://github.com/rapidsai/rapids-cmake/pull/633 is merged.

Comment (bdice) [reply cpp/cmake/rapids_config.cmake]: Done.

**Review by bdice — APPROVED** [review]

**Pull request merged**
